src/bosh_azure_cpi/lib/cloud/azure/models/disk_id.rb

```diff
@@ -26,103 +25,77 @@ class DiskId
     #  Parsing id for an existing disk
     #   disk_id = DiskId.parse(id, azure_config)
 
-    VERSION1 = 'v1' # class properties: version (string), id (string), default_resource_group_name (string)
-    VERSION2 = 'v2' # class properties: version (string), id (json)
+    CACHING_KEY = 'caching'
+    DISK_NAME_KEY = 'disk_name'
+    STORAGE_ACCOUNT_NAME_KEY = 'storage_account_name'
 
+    private_methods :_generate_data_disk_name
```

Comment:
what is purpose of `private_methods`?

Reply:
to make sure that nobody calls that out of this class.

Reply:
`private_methods` dost not work here. I guess should be `private_class_method`.